### ISSUE TITLE
NU2047 reports failure for new numeric types

### ISSUE THREAD

**Issue by stevenaw** [nunit/nunit.analyzers#1017]
Found while fixing https://github.com/nunit/nunit/issues/5340 via https://github.com/nunit/nunit/pull/5393

Some of the new numeric types report false positives for this when passed to `Within()`.

Sample repro

```csharp
[TestCaseSource(nameof(GetNumericToleranceValues))]
public void CanMatchAbsoluteTolerance<T>(T actualValue, T expectedValue)
    where T : INumber<T>
{
    Assert.That(actualValue, Is.EqualTo(expectedValue).Within(expectedValue), "Doesn't work with generic numeric constraints.");
    Assert.That(actualValue, Is.EqualTo((object)expectedValue).Within(expectedValue), "Doesn't work with object-based constraints.");
    Assert.That(actualValue, Is.EqualTo(expectedValue).Within(int.CreateChecked(expectedValue)), "Doesn't work with type variant tolerances.");
}

private static IEnumerable<TestCaseData> GetNumericToleranceValues()
{
    yield return new TestCaseData<Half>((Half)30, (Half)50);
    yield return new TestCaseData<Int128>((Int128)30, (Int128)50);
    yield return new TestCaseData<UInt128>((UInt128)30, (UInt128)50);
    yield return new TestCaseData<nint>((nint)30, (nint)50);
    yield return new TestCaseData<nuint>((nuint)30, (nuint)50);
}
```

**Comment by manfred-brands** [comment]
The issue here is the generics, the analyzer cannot know the type that will be passed in when checking the method.
It should ignore this if the type is not known at compile type.

**Comment by manfred-brands** [comment]
But even if the types are written out explicitly, the rule fires because it doesn't know about these types.
